### PR TITLE
test: adding unit tests for common

### PR DESCRIPTION
**Pull request by anandsunderraman**
**Description**

Adding unit tests to components

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
   "scripts": {
-    "test": "echo \"no test specified yet\"",
+    "test": "npm run testComponents",
     "release": "semantic-release",
     "lint": "eslint --max-warnings 0 --fix --config .eslintrc .",
     "generate:readme:toc": "markdown-toc -i README.md",
     "generate:assets": "npm run generate:readme:toc",
-    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION"
+    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION",
+    "testComponents": "rimraf test/temp && jest --coverage ./test/components"
```
Comment (derberg) [inline package.json:13]:
```suggestion
    "test:components": "rimraf test/temp && jest --coverage ./test/components"
```

small suggestion for script name, just to be consistent and follow the standards

File: package.json [hunk 2]
```diff
@@ -4,12 +4,13 @@
   "description": "Template for generating Go module based on async api",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"no test specified yet\"",
+    "test": "npm run testComponents",
```
Comment (derberg) [inline package.json:7]:
```suggestion
    "test": "npm run test:components",
```

File: package.json (hunk 1 above)
Comment (derberg) [inline package.json:13]: You do not put anything in `test/temp` so no need to remove it and using `rimraf`. I guess you saw it in other templates but this is related to not unit testing but a kind of integration testing where you actually run generator and then check if generated app matches the snapshot. Definitely another set of tests I recommend to add after this PR (example https://github.com/asyncapi/nodejs-template/pull/103)

File: test/components/common.test.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import { GetProtocolFlags, GetSubscriberFlags, pascalCase } from '../../components/common';
+import AsyncAPIDocument from '@asyncapi/parser/lib/models/asyncapi';
+
+const docWithoutChannels = new AsyncAPIDocument({
```
Comment (derberg) [inline test/components/common.test.js:4]: I recommend to take these example files from test into separate files, and then just read them from file. This way you can share them in future with others tests, like integration tests

**Review by derberg — COMMENTED** [review]
looks good, left just few recommendations

File: package.json [hunk 4]
```diff
@@ -33,7 +34,8 @@
     "@asyncapi/generator-filters": "^2.0.0",
     "@asyncapi/generator-hooks": "^0.1.0",
     "@asyncapi/generator-react-sdk": "^0.2.23",
-    "@asyncapi/modelina": "^0.45.0"
+    "@asyncapi/modelina": "^0.45.0",
+    "rimraf": "^3.0.2"
```
Comment (derberg) [inline package.json:38]: you do not use it, you can remove and update package-lock

**Review by derberg — DISMISSED** [review]
looks very good, just remove `rimraf` and yeah heading on to @smoya to approve as my approval doesn't count here

**Review by smoya — APPROVED** [review]
LGTM! 👍 🚀

**Pull request merged**